main.c

```diff
@@ -624,6 +633,12 @@ int main(int argc, char **argv) {
   SetWMProperties(display, saver_window, "xsecurelock", "saver", argc, argv);
   my_windows[n_my_windows++] = saver_window;
 
+  Window auth_window =
+      XCreateWindow(display, saver_window, 0, 0, w, h, 0, CopyFromParent,
```

Comment:
Note that you have no code to update the size of the auth_window after creation, so after attaching/detaching a screen its size will be incorrect.

I see that this is intentional so the auth window does not have to cover everything, so I won't suggest updating its size in the loop below. But can you change auth_x11 to resize the auth window to match the screen size? With that this should be good.

Reply:
Done. I had to use monitor change instead of listening for structure change on root_window, but that should be fine.

Reply:
LGTM.